external/radiation/radiation/wrapper_api.py

```diff
@@ -247,13 +246,13 @@ def _rad_compute(self, state: State, time: cftime.DatetimeJulian,) -> Diagnostic
             self._cached = unstack(postprocess_out(out), coords)
         return self._cached
 
-
-def _solar_hour(time: cftime.DatetimeJulian) -> float:
-    return (
-        time.hour
-        + time.minute / MINUTES_PER_HOUR
-        + time.second / (MINUTES_PER_HOUR * SECONDS_PER_MINUTE)
-    )
+    def _solar_hour(self, time: cftime.DatetimeJulian) -> float:
+        """This follows the Fortran computation that rounds initial time to the
+        hour. See https://github.com/NOAA-GFDL/SHiELD_physics/issues/17, but for
+        now we want the port to validate against Fortran."""
+        seconds_elapsed = (time - self._init_time).total_seconds()
+        hours_elapsed = seconds_elapsed / SECONDS_PER_HOUR
+        return hours_elapsed + self._init_time.hour
```

Comment:
Whoops this should be:
```suggestion
return (hours_elapsed + self._init_time.hour) % 24
```
Not sure if that would have caused an error or not with periodic cosine zenith angle computations, but will fix.